augur/api/view/routes.py

```diff
@@ -213,7 +188,95 @@ def user_delete():
 @app.route('/account/settings')
 @login_required
 def user_settings():
-    return render_template("settings.j2")
+    return render_template('settings.j2')
+
+
+@app.route('/account/api-keys')
+@login_required
+def api_keys_list():
+    """
+    Get a list of API keys for the current user.
+    """
+    try:
+        # Query the client applications for the current user to get their API keys
+        applications = db_session.query(ClientApplication).filter(
+            ClientApplication.user_id == current_user.user_id
+        ).all()
+        
+        keys = []
+        for app in applications:
+            keys.append({
+                'id': app.id,
+                'api_key': app.api_key,
+                'name': app.name,
+                'created_at': app.redirect_url  # Using redirect_url as created_at for demonstration
+            })
+        
+        return jsonify({'success': True, 'keys': keys})
+    except (ValueError, AttributeError, RuntimeError) as e:
+        logger.error(f"Error getting API keys: {str(e)}")
+        return jsonify({'success': False, 'message': str(e)})
+
+
+@app.route('/account/generate-api-key', methods=['POST'])
+@login_required
+def generate_api_key():
+    """
+    Generate a new API key for the current user.
+    """
+    try:
+        # Generate a new API key
+        new_key = secrets.token_hex(16)
+        
+        # Create a new client application with the generated API key
+        new_app = ClientApplication(
+            id=secrets.token_hex(16),
+            api_key=new_key,
+            name=f"API Key - {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}",
+            redirect_url=datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
+            user_id=current_user.user_id
+        )
+        
+        # Add the new application to the database
+        db_session.add(new_app)
+        db_session.commit()
+        
+        return jsonify({'success': True, 'api_key': new_key})
+    except (ValueError, AttributeError, RuntimeError) as e:
+        db_session.rollback()
+        logger.error(f"Error generating API key: {str(e)}")
+        return jsonify({'success': False, 'message': str(e)})
+
+
+@app.route('/account/delete-api-key', methods=['POST'])
+@login_required
+def delete_api_key():
```

Comment:
Endpoints returning a non-view response should not go in this file, all routes in this file must return a view. Routes returning JSON data or any other non-view components should be placed in either `augur.api.view.api` or `augur.api.routes`.